docs/dnssec/pkcs11.rst

```diff
@@ -40,6 +40,67 @@ These instructions have been tested on Debian 10 (Buster).
 
     pdnsutil show-zone example.com
 
+SoftHSM2 with forwarding
+------------------------
+
+Based on https://p11-glue.github.io/p11-glue/p11-kit/manual/remoting.html.
+
+You need to install ``gnutls-bin`` to get token URLs.
+
+You cannot run ``p11-kit server`` as root, so you will need some user for running it. This user must belong to ``softhsm`` group.
+
+These commands need to be ran as the non-root user (we shall call it tokenuser)
+
+First, setup your token::
+
+   softhsm2-util --init-token --label "ecdsa#1" --pin 1234 --so-pin 1234 --free
+   pkcs11-tool --module /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so --keypairgen --key-type EC:prime256v1 --pin 1234 -a 'my key'
```

Comment:
How does this command know it should use the token you inited one line earlier? Am I missing something?

Reply:
Well, it works if you only have one slot in use. But I guess I could figure out the syntax for specifying slot too.